azurelinuxagent/common/protocol/wire.py

```diff
@@ -1094,7 +1086,6 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
         return ret
 
     def upload_status_blob(self):
-        self.update_goal_state()
         ext_conf = self.get_ext_conf()
 
         if ext_conf.status_upload_blob is None:
```

Comment:
(github won't let me expand and comment on L1093), but why are we keeping that call to update the goal state? Because it's forced and otherwise we don't force it from the main loop? Do we even need a force update here?

Reply:
The status blob is in extensions config, so a force update would be needed if this condition was true.

As far as the if statement itself, I don't see why it is needed and I may remove it as we continue cleaning the code.